### PR TITLE
Update previous terraform.io doc links to updated developer.hashicorp doc links

### PR DESCRIPTION
**Pull request by Uk1288**
## Description
This PR has following changes:
- Update old terraform.io api docs links to the new developer.hashicorp api doc links
-  Update non-working links

## Output from tests
Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.

```
$ TFE_ADDRESS="https://example" TFE_TOKEN="example" go test ./... -v -run TestFunctionsAffectedByChange

...
```

### CODE REVIEW

**Review by brandonc — DISMISSED** [review]
🥇Thank you for this cleanup!

File: docs/TESTS.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ You'll need to have environment variables setup in your environment to run the t
 Tests are run against an actual backend so they require a valid backend address and token:
 
 1. `TFE_ADDRESS` - URL of a Terraform Cloud or Terraform Enterprise instance to be used for testing, including scheme. Example: `https://tfe.local`
-1. `TFE_TOKEN` - A [user API token](https://www.terraform.io/docs/cloud/users-teams-organizations/users.html#api-tokens) for the Terraform Cloud or Terraform Enterprise instance being used for testing.
+1. `TFE_TOKEN` - A [user API token](https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/users#api-tokens) for the Terraform Cloud or Terraform Enterprise instance being used for testing.
```
Comment (nfagerlund) [inline docs/TESTS.md:42]:
```suggestion
1. `TFE_TOKEN` - A [user API token](https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/users#tokens) for the Terraform Cloud or Terraform Enterprise instance being used for testing.
```

The main path is fine but the in-page anchor changed at some point.

File: run_task.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ var _ RunTasks = (*runTasks)(nil)
 
 // RunTasks represents all the run task related methods in the context of an organization
 // that the Terraform Cloud/Enterprise API supports.
-// https://www.terraform.io/cloud-docs/api-docs/run-tasks/run-tasks#run-tasks-api
+// https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run-tasks/run-tasks#run-tasks-api
```
Comment (nfagerlund) [inline run_task.go:14]:
```suggestion
// https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run-tasks/run-tasks
```

**Review by nfagerlund — APPROVED** [review]
Looks great! Caught a couple optional anchor adjustments to make, but then got tired of scanning all these links.

**Review by nfagerlund — APPROVED** [review]

**Pull request merged**